SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -802,7 +802,7 @@ TWeakObjectPtr<AActor> ASpatialDebugger::GetHitActor()
 	return HitActors[HoverIndex];
 }
 
-FVector2D ASpatialDebugger::ProjectActorToScreen(const TWeakObjectPtr<AActor> Actor, const FVector& PlayerLocation)
+FVector2D ASpatialDebugger::ProjectActorToScreen(const TWeakObjectPtr<AActor> Actor, const FVector& PlayerLocation, const UCanvas* Canvas)
```

Comment:
I would suggest simplifying it to the following:
```
FVector2D ASpatialDebugger::ProjectActorLocationToScreen(const FVector& ActorLocation, const FVector& PlayerLocation, const UCanvas* Canvas)
{
	if (ActorLocation.IsZero() || FVector::Dist(PlayerLocation, ActorLocation) > MaxRange || !LocalPlayerController.IsValid())
	{
		return FVector2D::ZeroVector;
	}

	SCOPE_CYCLE_COUNTER(STAT_Projection);
	return FVector2D(Canvas->Project(ActorLocation + WorldSpaceActorTagOffset));
}
```

Reply:
Actually I prefer this. The use of FVector2D::ZeroVector as sentinel value is not super nice I think also has a minor bug, what if the correct projection coordinate is actually 0,0 we will not draw it in that case.
```
bool ASpatialDebugger::ProjectActorLocationToScreen(const FVector& ActorLocation, const FVector& PlayerLocation, const FVector& OutLocation, const UCanvas* Canvas)
{
	if (ActorLocation.IsZero() || FVector::Dist(PlayerLocation, ActorLocation) > MaxRange)
	{
		return false;
	}

	SCOPE_CYCLE_COUNTER(STAT_Projection);
	OutLocation = FVector2D(Canvas->Project(ActorLocation + WorldSpaceActorTagOffset));
	return true;
}
```

Reply:
See other comments for how it would look in the rest of the code.